### PR TITLE
Chunked upload endpoints

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: boxsdk/object/__init__.py [hunk 1]
```diff
@@ -5,4 +5,14 @@
...
+__all__ = list(map(str, [
+    'collaboration',
+    'events',
+    'file',
+    'folder',
+    'group',
+    'group_membership',
+    'search',
+    'upload_session',
+    'user'
```
Comment (SajadMirzaei) [inline boxsdk/object/__init__.py]: trailing comma

File: boxsdk/object/upload_session.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class UploadSession(BaseObject):
+    _item_type = 'upload_session'
+    _parent_item_type = 'file'
+
+    def get_url(self, *args):
+        """
+        Base class override. Endpoint is a little different - it's /files/upload_sessions.
+        """
```
Comment (SajadMirzaei) [inline boxsdk/object/upload_session.py]: return type missing in doc

File: boxsdk/object/upload_session.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    def get_parts(self):
+        """
+        Get a list of parts uploaded so far.
+
+        :rtype:     `list` of `dict`
+        """
+        response = self.session.get(self.get_url('parts')).json()
+        return response['entries']
+
+    def _calculate_part_sha1(self, content_stream):
```
Comment (SajadMirzaei) [inline boxsdk/object/upload_session.py]: doc missing

File: boxsdk/object/folder.py [hunk 4]
```diff
@@ -419,3 +419,19 @@ def delete(self, recursive=True, etag=None):
         :raises: :class:`BoxAPIException` if the specified etag doesn't match the latest version of the folder.
         """
         return super(Folder, self).delete({'recursive': recursive}, etag)
+
+    def create_upload_session(self, file_size, file_name):
+        """
+        Create a new chunked upload session for uploading a new file.
+         :param file_size:       The size of the file that will be uploaded.
```
Comment (anthonywee) [inline boxsdk/object/folder.py]: extra space

File: boxsdk/object/file.py [hunk 5]
```diff
@@ -2,9 +2,12 @@
 
 from __future__ import unicode_literals
 
+import json
```
Comment (anthonywee) [inline boxsdk/object/file.py]: could add absolute_import above

**Pull request closed without merging**